packages/pluggableWidgets/screenshottaker-native/src/ScreenshotTaker.tsx

```diff
@@ -0,0 +1,115 @@
+import { flattenStyles } from "@mendix/piw-native-utils-internal";
+import { createElement, ReactElement, useCallback, useEffect, useRef, useState } from "react";
+import { View, Text } from "react-native";
+import { ValueStatus } from "mendix";
+import { captureScreen, captureRef, CaptureOptions } from "react-native-view-shot";
+
+import { executeAction } from "@mendix/piw-utils-internal";
+
+import { ScreenshotTakerProps } from "../typings/ScreenshotTakerProps";
+import { defaultScreenshotTakerStyles, ScreenshotTakerType } from "./ui/Styles";
+
+declare type CustomGlobal = NodeJS.Global &
+    typeof globalThis & {
+        screenshotRunner: {
+            masterIsRunning: boolean;
+            task: {
+                name: string;
+                isRunning: boolean;
+                readyToTakeScreenshot: boolean;
+                error: boolean;
+                onError: (e: Error, name: string) => any;
+            };
+        };
+    };
+
+const timeout = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
+
+export function ScreenshotTaker(props: ScreenshotTakerProps<ScreenshotTakerType>): ReactElement | null {
+    const captureContent = props.whatToCapture === "captureContent";
+    const viewRef = useRef(null);
+    const shouldTakeScreenshotRef = useRef(false);
+    const [shouldTakeScreenshot, setShouldTakeScreenshot] = useState(false);
+    const styles = flattenStyles(defaultScreenshotTakerStyles, props.style);
+    const g = global as CustomGlobal;
+
+    // Runs on Unmount
+    useEffect(() => {
+        return () => {
+            if (g.screenshotRunner?.task) {
+                (global as CustomGlobal).screenshotRunner.task.isRunning = false;
+            }
+        };
+    }, []);
+
+    // Runs on dependency change
+    useEffect(() => {
+        if (
+            (global as CustomGlobal).screenshotRunner?.task?.isRunning &&
+            !shouldTakeScreenshotRef.current &&
+            !shouldTakeScreenshot
+        ) {
+            shouldTakeScreenshotRef.current = true;
+            setShouldTakeScreenshot(true);
+        }
+    }, [shouldTakeScreenshotRef.current, setShouldTakeScreenshot]);
```

Comment:
`setShouldTakeScreenshot` should be replaced with `shouldTakeScreenshot`

Reply:
done